public/jira/derived/work-timing/work-timing.ts

```diff
@@ -0,0 +1,235 @@
+import { getBusinessDatesCount } from "../../../status-helpers.js";
+import {
+  estimateExtraPoints,
+  sampleExtraPoints,
+} from "../../../shared/confidence.js";
+import {
+  DueData,
+  getStartDateAndDueDataFromFieldsOrSprints,
+  getStartDateAndDueDataFromSprints,
+  StartData,
+} from "../../../shared/issue-data/date-data.js";
+import {
+  DefaultsToConfig,
+  NormalizedIssue,
+  NormalizedTeam,
+} from "../../shared/types.js";
+
+export type DerivedWorkTiming = {
+  isConfidenceValid: boolean;
+  usedConfidence: number;
+  isStoryPointsValid: boolean;
+  defaultOrStoryPoints: number;
+  storyPointsDaysOfWork: number;
+  isStoryPointsMedianValid: boolean;
+  defaultOrStoryPointsMedian: number;
+  storyPointsMedianDaysOfWork: number;
+  deterministicExtraPoints: number;
+  deterministicExtraDaysOfWork: number;
+  deterministicTotalPoints: number;
+  deterministicTotalDaysOfWork: number;
+  probablisticExtraPoints: number;
+  probablisticExtraDaysOfWork: number;
+  probablisticTotalPoints: number;
+  probablisticTotalDaysOfWork: number;
+  hasStartAndDueDate: boolean;
+  startAndDueDateDaysOfWork: number | null;
+  hasSprintStartAndEndDate: boolean;
+  sprintDaysOfWork: number | null;
+  sprintStartData: StartData;
+  endSprintData: DueData;
+  totalDaysOfWork: number | null;
+  defaultOrTotalDaysOfWork: number | null;
+  completedDaysOfWork: number;
+} & StartData &
+  DueData;
+
+/**
+ * @param {NormalizedTeam} team
+ * @returns {number}
+ */
+export function getConfidenceDefault(team: NormalizedTeam) {
+  return 50;
+}
+
+/**
+ *
+ * @param {NormalizedTeam} team
+ * @returns number
+ */
+export function getStoryPointsDefault(team: NormalizedTeam) {
+  return team.velocity / team.parallelWorkLimit;
+}
+
+const defaults = {
+  getConfidenceDefault,
+  getStoryPointsDefault,
+};
+
+export type WorkTimingConfig = DefaultsToConfig<typeof defaults>;
+
+/**
+ *
+ * @param {import("../../shared/types.js").NormalizedIssue} normalizedIssue
+ * @param {*} param1
+ * @returns {DerivedWorkTiming}
+ */
+export function deriveWorkTiming(
+  normalizedIssue: NormalizedIssue,
+  {
+    getConfidence = getConfidenceDefault,
+    getStoryPoints = getStoryPointsDefault,
+    uncertaintyWeight = 80,
+  }: Partial<WorkTimingConfig> & { uncertaintyWeight?: number } = {}
+) {
```

Comment:
do we need a return type?

edit: After looking at the previous and rest of the changes, we should probably take a stance on whether we add return types or not. Generally, I prefer to explicitly type return types on functions that are exported unless there's a reason not to... What are your thoughts?

Reply:
I added explicit return types for all functions